pkg/validate/validate_test.go

```diff
@@ -324,3 +323,114 @@ func TestValidate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateUpdateContainerService(t *testing.T) {
+
+	var twoAgents = []byte(`---
+location: eastus
+name: openshift
+properties:
+  routerProfiles:
+  - name: default
+    publicSubdomain: test.example.com
+  agentPoolProfiles:
+  - name: infra
+    role: infra
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: myCompute
+    role: compute
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+`)
+	var threeAgents = []byte(`---
+location: eastus
+name: openshift
+properties:
+  routerProfiles:
+  - name: default
+    publicSubdomain: test.example.com
+  agentPoolProfiles:
+  - name: infra
+    role: infra
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: myCompute
+    role: compute
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: mySomethingElse
+    role: compute
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+`)
+
+	tests := map[string]struct {
+		newTemplate   []byte
+		newAgentCount int
+		oldTemplate   []byte
+		oldAgentCount int
+		externalOnly  bool
```

Comment:
would remove and just hard tie the validateUpdateContainerService() argument to false for now